lib/Document/Adapter/LibreOffice.php

```diff
@@ -169,12 +169,9 @@ public function getText(?int $page = null, ?Asset\Document $asset = null, ?strin
         try {
             if (!parent::isFileTypeSupported($asset->getFilename())) {
                 $file = $this->getPdf($asset);
-                if (!is_resource($file)) {
```

Comment:
`getPdf()` might be returning `null` or is  it a false positive from PhpStorm?

![image](https://github.com/pimcore/pimcore/assets/6014195/cd7549cf-0af4-4f22-9e48-beaa81cf0468)

Reply:
@kingjia90 The php doc say it is always a resource:
https://github.com/pimcore/pimcore/blob/cfa3ba70a0eb5794277a242ebc4396fae7f8b898/lib/Document/Adapter.php#L53
So we have to change the php doc?

Reply:
The getText() methods also use the getPdf() method and doesn't check for null. So we have to add there also a check. Or the getPdf() should throw the exception so we don't need it everywhere.

Reply:
See https://github.com/pimcore/pimcore/pull/14799/commits/e0b59a8cdfd81e0828d1cec47362910c3674b914 
Now it should always return a resource or throw an exception